src/FormatNumber.elm

```diff
@@ -37,115 +95,61 @@ type alias Locale =
     >>> formatFloat (Locale 1 "," ".") -0.01
     "0.0"
 
+    >>> formatFloat (Locale 2 "," ".") 0.01
+    "0.01"
+
+    >>> formatFloat (Locale 0 "," ".") 123.456
+    "123"
+
+    >>> formatFloat (Locale 0 "," ".") 1e9
+    "1,000,000,000"
+
+    >>> formatFloat (Locale 5 "," ".") 1.0
+    "1.00000"
 -}
 formatFloat : Locale -> Float -> String
 formatFloat locale num =
-    let
-        multiplier : Int
-        multiplier =
-            10 ^ locale.decimals
-
-        digits : Int
-        digits =
-            num
-                |> (*) (toFloat multiplier)
-                |> round
-    in
-        formattedNumber locale digits
+    (formatInt locale (truncate num))
+        ++ (separator locale)
+        ++ (nDigits locale.decimals num)
 
 
 {-| Format a integer number as a pretty string:
 
     >>> formatInt { decimals = 1, thousandSeparator = ",", decimalSeparator = "." } 0
-    "0.0"
+    "0"
```

Comment:
> The user doesnt explicitly asked for a number of decimals. It's just part of the locale.

It's part of the `Locale` user coded or choose, ie. a choice.

> My opinion is that integers should never be printed with decimals

I totally disagree. Prices are usually expressed with cents (or pences, or whatever), for example. Formally speaking, for example, it's _$ 1,00_, not _$ 1_ (even if informally we use _$ 1_). I still believe adding an option in the `Locale` is a better choice.